vars/folioCypressFlow.groovy

```diff
@@ -21,174 +21,174 @@ import org.folio.Constants
-    String adminUsername = params.adminUsername
-    String adminPassword = params.adminPassword
-    String parallelExecParameters = params.parallelExecParameters
-    String sequentialExecParameters = params.sequentialExecParameters
-    String testsTimeout = params.testsTimeout?.trim() ?: Constants.GLOBAL_BUILD_TIMEOUT
-    String testrailProjectID = params.testrailProjectID
-    String testrailRunID = params.testrailRunID
-    int numberOfWorkers = params.numberOfWorkers as int ?: 1
-    String agent = params.agent
-    String browserName = "chrome"
-    String cypressImageVersion = ''
-    List resultPaths = []
-
-    buildName customBuildName
-
-
-    timeout(time: testsTimeout, unit: 'HOURS') {
-        if (parallelExecParameters?.trim()) {
-            stage('[Cypress] Parallel run') {
-                script {
-                    Map workers = [:]
-                    for (int workerNumber = 1; workerNumber <= numberOfWorkers; workerNumber++) {
-                        workers["Worker#${workerNumber}"] = { currentWorkerNumber ->
-                            node('rancher||jenkins-agent-java11') {
-                                cloneCypressRepo(branch)
-
-                                cypressImageVersion = getCypressImageVersion()
-
-                                executeTests(cypressImageVersion, tenantUrl, okapiUrl, tenantId, adminUsername, adminPassword,
-                                  customBuildName, browserName, parallelExecParameters, testrailProjectID, testrailRunID, true)
-
-                                resultPaths.add(archiveTestResults(currentWorkerNumber))
-                            }
-                        }.curry(workerNumber)
-                    }
-                    parallel(workers)
-                }
-            }
+  folioTools.validateParams(params, ['parallelExecParameters', 'sequentialExecParameters', 'testrailProjectID', 'testrailRunID', 'numberOfWorkers'])
+
+  /* Define variables */
+  String customBuildName = params.customBuildName?.trim() ?
+    "${params.customBuildName.replaceAll(/[^A-Za-z0-9\s.]/, "").replace(' ', '_')}.${env.BUILD_ID}" : env.BUILD_ID
+  String branch = params.branch
+  String tenantUrl = params.tenantUrl
+  String okapiUrl = params.okapiUrl
+  String tenantId = params.tenantId
+  String adminUsername = params.adminUsername
+  String adminPassword = params.adminPassword
+  String parallelExecParameters = params.parallelExecParameters
+  String sequentialExecParameters = params.sequentialExecParameters
+  String testsTimeout = params.testsTimeout?.trim() ?: Constants.GLOBAL_BUILD_TIMEOUT
+  String testrailProjectID = params.testrailProjectID
+  String testrailRunID = params.testrailRunID
+  int numberOfWorkers = params.numberOfWorkers as int ?: 1
+  String agent = params.agent
+  String browserName = "chrome"
+  String cypressImageVersion = ''
+  List resultPaths = []
+
+  buildName customBuildName
+
+
+  timeout(time: testsTimeout, unit: 'HOURS') {
```

Comment:
Should it(timeout) be under options {} section?

Reply:
No as we restrict with timeout exact stage related to tests execution